src/transaction/async_log_page_receiver.cpp

```diff
@@ -0,0 +1,49 @@
+#include "async_log_page_receiver.hpp"
+
+namespace cublog
+{
+  async_log_page_receiver::async_log_page_receiver (/* args */)
+  {
+  }
+
+  async_log_page_receiver::~async_log_page_receiver ()
+  {
+  }
+
+  void
+  async_log_page_receiver::set_page_requested (LOG_PAGEID log_pageid)
+  {
+    std::unique_lock<std::mutex> lock (m_reqested_pages_mutex);
+    m_requested_page_ids.insert (log_pageid);
+  }
+
+  bool
+  async_log_page_receiver::is_page_requested (LOG_PAGEID log_pageid)
+  {
+    std::unique_lock<std::mutex> lock (m_reqested_pages_mutex);
+    return m_requested_page_ids.find (log_pageid) != m_requested_page_ids.end ();
```

Comment:
Requested page and received pages are never erased. Once all active requests have been served, the request and the page should be removed.